File: VBdotNet2010/DocxPDFTIFFConverter/Form1.vb
Public Class Form1

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        OpenFileDialog1.Filter = "PDF (*.pdf)|*.pdf|TIF (*.tif)|*.tif||"




        If OpenFileDialog1.ShowDialog() = DialogResult.OK Then
            txtimagefile.Text = OpenFileDialog1.FileName

        End If


    End Sub

    Private Sub Button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button2.Click
        Dim strFile As String


        strFile = Me.txtimagefile.Text

        If strFile = "" Then

            MessageBox.Show("Please select the image")
            Exit Sub
        End If

        strFile.ToLower()

        If strFile.Substring(strFile.Length() - 3) = "pdf" Then


            If Me.AxImageViewer1.DocxPDF2Docx(txtimagefile.Text, "c:\test1.docx") Then
                MessageBox.Show("Save c:\test1.docx completed")
                Exit Sub
            End If

        Else
            If Me.AxImageViewer1.DocxTIFF2Docx(txtimagefile.Text, "c:\test1.docx") Then
                MessageBox.Show("Save c:\test1.docx completed")
                Exit Sub
            End If


        End If

    End Sub

    Private Sub chkuseadvpdf_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles chkuseadvpdf.CheckedChanged
        If chkuseadvpdf.Checked Then
            Me.AxImageViewer1.PDFUseAdvancedViewer = True
        Else
            Me.AxImageViewer1.PDFUseAdvancedViewer = False

        End If

    End Sub
End Class

File: c#2019/DocxPDFTIFFConverter/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace WindowsFormsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            this.openFileDialog1.Filter = "PDF (*.pdf)|*.pdf|Tif (*.tif)|*.tif";
            if (this.openFileDialog1.ShowDialog(this) == DialogResult.OK)
             {

                 textImage.Text= this.openFileDialog1.FileName;

             }
        }

        private void button2_Click(object sender, EventArgs e)
        {

            
            string strImage = textImage.Text;

            if (strImage == "")
            {
                MessageBox.Show("Please select the image");
                return;
            }
            strImage.ToLower();

                if (strImage.Substring(strImage.Length - 3) == "pdf")
                {
                    if (axImageViewer1.DocxPDF2Docx(strImage, "c:\\test1.docx"))
                        MessageBox.Show("c:\\test1.docx completed");

                }
                else
                {
                    if (axImageViewer1.DocxTIFF2Docx(strImage, "c:\\test1.docx"))
                        MessageBox.Show("c:\\test1.docx completed");


                }

        }

        private void chkuseadvpdf_CheckedChanged(object sender, EventArgs e)
        {
            if (chkuseadvpdf.Checked)
                this.axImageViewer1.PDFUseAdvancedViewer = true;
            else
                this.axImageViewer1.PDFUseAdvancedViewer = false;
     
        }
    }
}
